src/ig/IGExporter.ts

```diff
@@ -330,68 +305,41 @@ export class IGExporter {
-          });
-          return;
+        if (
+          (!existsSync(inputIndexMarkdownPagesPath) && existsSync(inputIndexXMLPagesPath)) ||
+          (!existsSync(inputIndexMarkdownPageContentPath) &&
+            existsSync(inputIndexXMLPageContentPath))
+        ) {
+          generation = 'html';
         }
+        // Add user-provided index file to IG definition
+        this.ig.definition.page.page.push({
+          nameUrl: 'index.html',
+          title: 'Home',
+          generation
+        });
+        return;
       }
 
-      const base = this.isIgPubContext ? '' : path.basename(this.igDataPath);
-      const filePathString = path.join(base, 'input');
+      // Generate index file if indexPageContent is provided in config
+      const filePathString = 'input';
       const warning = warningBlock('<!-- index.md {% comment %}', '{% endcomment %} -->', [
         `To change the contents of this file, edit the "indexPageContent" attribute in the tank ${this.configName} file`,
         `or provide your own index file in the ${filePathString}${path.sep}pagecontent or ${filePathString}${path.sep}pages folder.`,
         'See: https://build.fhir.org/ig/FHIR/ig-guidance/using-templates.html#root.input'
       ]);
       const outputPath = path.join(pageContentExportPath, 'index.md');
       outputFileSync(outputPath, `${warning}${this.config.indexPageContent}`);
-      this.updateOutputLog(outputPath, [this.configPath], 'generated');
       logger.info(`Generated index.md based on "indexPageContent" in ${this.configName}.`);
-    } else if (existsSync(inputIndexMarkdownPageContentPath)) {
-      if (this.shouldCopyFiles) {
-        ensureDirSync(pageContentExportPath);
-        this.copyWithWarningText(
-          inputIndexMarkdownPageContentPath,
-          path.join(pageContentExportPath, 'index.md')
-        );
-        logger.info(`Copied ${path.join(pageContentExportPath, 'index.md')}`);
-      }
-    } else if (existsSync(inputIndexXMLPageContentPath)) {
+    } else if (
+      !existsSync(inputIndexMarkdownPageContentPath) &&
+      existsSync(inputIndexXMLPageContentPath)
+    ) {
       generation = 'html';
-      if (this.shouldCopyFiles) {
-        ensureDirSync(pageContentExportPath);
-        this.copyWithWarningText(
-          inputIndexXMLPageContentPath,
-          path.join(pageContentExportPath, 'index.xml')
-        );
-        logger.info(`Copied ${path.join(pageContentExportPath, 'index.xml')}`);
-      }
-    } else if (existsSync(inputIndexMarkdownPagesPath)) {
-      if (this.shouldCopyFiles) {
-        ensureDirSync(pagesExportPath);
-        this.copyWithWarningText(
-          inputIndexMarkdownPagesPath,
-          path.join(pagesExportPath, 'index.md')
-        );
-        logger.info(`Copied ${path.join(pagesExportPath, 'index.md')}`);
-      }
-    } else if (existsSync(inputIndexXMLPagesPath)) {
+    } else if (!existsSync(inputIndexMarkdownPagesPath) && existsSync(inputIndexXMLPagesPath)) {
```

Comment:
Couldn't this `else if` be combined with the one above so that it is the same as the one on line 308?

Reply:
Yep! Updated in e02ff58